### DISCUSSION TITLE
Is there way emit everyone except sender/"person who made request" inside express route

### DISCUSSION THREAD

**Discussion by JuHaNi654** [socketio/socket.io#4087] [Q&A] [answered]
Hi,
I was wondering is there way emitting everyone except sender inside express route? 

`router.post('/test', (req, res) => {
   const io = req.app.get('io')
    //  Will send everyone including to the person who made request
   io.to("room").emit('txt', data)
    //  Will also send everyone including to the person who made request
   io.socket.to("room").emit('txt', data)
  ...
}) `

**Comment by darrachequesne** [comment, marked as answer]
It should be doable with express-session:

```js
io.on("connection", (socket) => {
  // join the room identified by the session ID
  socket.join(socket.request.sessionID);
});

router.post("/test", (req, res) => {
  // send to everyone except those in the given room
  io.except(req.sessionID).emit(/* ... */);
});
```

For the setup, please see https://socket.io/docs/v4/faq/#usage-with-express-session (until we have a proper example with express-session)

**Reply by JuHaNi654** [reply]
Alright thx 👍.
